lib/collection/src/collection/shard_transfer.rs

```diff
@@ -314,11 +325,22 @@ impl Collection {
         }
 
         if transfer.from == self.this_peer_id {
-            transfer::driver::revert_proxy_shard_to_local(shard_holder, transfer.shard_id).await?;
+            transfer::driver::revert_proxy_shard_to_local(&shard_holder, transfer.shard_id).await?;
         }
 
         shard_holder.register_abort_transfer(&transfer_key)?;
 
+        if is_resharding_transfer {
+            let resharding_state = shard_holder.resharding_state.read().clone();
+
+            // `abort_resharding` locks `shard_holder`!
+            drop(shard_holder);
+
+            if let Some(state) = resharding_state {
+                self.abort_resharding(state.key(), false).await?;
+            }
+        }
```

Comment:
How about removing the clone and just taking the key. Then we can drop the shard holder and pass it into the abort function.

Obviously a nitpick, but I noticed this when messing with the code.